context/app/static/js/components/workspaces/NewWorkspaceDialog/useCreateWorkspaceForm.ts

```diff
@@ -77,8 +111,10 @@ function useCreateWorkspaceForm({ defaultName, defaultTemplate, initialProtected
         'protected-datasets': checkedProtectedDatasets,
         templates: [defaultTemplate ?? DEFAULT_TEMPLATE_KEY],
         workspaceJobTypeId: DEFAULT_JOB_TYPE,
+        datasets: initialSelectedDatasets ?? [],
       });
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I generally try to avoid disabling this lint rule in particular. Breaking the rules of react hooks leads to annoying bugs like the one we're working around with that forced refresh approach. 

Essentially, since `initialSelectedDatasets` isn't part of the dependency array, if its value changes upstream, it still uses the previous value when running this effect due to `initialSelectedDatasets` being an object.

Maybe the `initialSelectedDatasets` not being equal to `checkedSelectedDatasets` and thus causing the refresh cycle is something to investigate? i.e. if `initialSelectedDatasets` is undefined and nullishly coalesces to a new instance of an empty array, that wouldn't be instantiated in the same spot in memory and would thus fail an equality check. 

Another idea: if this effect is meant to fire when there are initial protected datasets, would it be possible to have this reset occur as part of the `remove protected datasets` event handler instead of running it as a separate effect? That would also resolve the infinite render cycle issue you encountered, to my understanding. See: https://react.dev/learn/you-might-not-need-an-effect

A third idea - wrapping a value in a `ref` is a useful escape hatch when you're absolutely certain you don't want updates to said value to trigger the useEffect.

Reply:
Thanks for the advice! This was helpful, I worked from your second suggestion - while working on this I forgot that the original intention of this `useEffect` was just to set `initialProtectedDatasets` after dataset selection. Instead of resetting the entire form, I changed to just setting the value of `initialProtectedDatasets` directly.

Reply:
One note - this does not resolve the issue of the forced re-render in the `AddDatasetsTable/hooks.ts`.